UMLEditor/UMLEditor/Classes/Diagram.cs

```diff
@@ -222,9 +230,10 @@ public void AddMethod(string toClass, string returnType, string methodName, List
     /// <exception cref="ClassAlreadyExistsException">Ensures there is not already a class by this name</exception>
     public void AddClass(string className)
     {
+
         if (ClassExists(className))
         {
-            throw new ClassAlreadyExistsException($"Class {className} already exists");
+            throw new ClassNonexistentException($"Class '{className}' already exists");
```

Comment:
Shouldn't we be throwing a ClassAlreadyExistsException here? We are checking if the class exists (because you cannot create duplicate classes).